client/rest/src/main/java/org/elasticsearch/client/RestClientView.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+
+/**
+ * Stateless view into a {@link RestClient} with customized parameters.
+ */
+class RestClientView extends AbstractRestClientActions {
```

Comment:
I'm not really happy with any of the names.

Reply:
I'd talked about this with @javanna a while ago and we're kind of abandoning this approach because we think it is too confusing. We won't have a view, I think.

Reply:
Maybe the fact we have a good use case will help make a case for it :)

BTW I would just need the interface: I believe re-defining the interface just to remove some methods in Hibernate is not ideal as it would imply a high coupling, while I'd like to expose your interface so that people using the latest Elasticsearch would have any new features immediately, w/o us having to update our facade first. (Normally we'd have a facade to better maintain the API but this is for advanced users, which will prefer to not have anything in the way)